x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkAction.java

```diff
@@ -173,10 +169,26 @@ public boolean isSafeAction() {
                         indexMetadata.getIndex().getName(),
                         indexMetadata.getLifecyclePolicyName()
                     );
-                    return true;
+                    listener.onResponse(true);
+                    return;
                 }
-                return false;
-            }
+                String indexName = indexMetadata.getIndex().getName();
+                client.admin()
+                    .indices()
+                    .prepareStats(indexName)
+                    .clear()
+                    .setDocs(true)
+                    .setStore(true)
+                    .execute(listener.delegateFailure((delegateListener, indicesStatsResponse) -> {
```

Comment:
can we add a test for this such that it fails and moves ILM to the ERROR step?

Reply:
I added a test that checks that the error gets propagated to the listener which I think is effectively the same, because the `onFailure` of the `IndexLifecycleRunner` will move it to the error step.